tss-esapi/src/context/general_esys_tr.rs

```diff
@@ -388,6 +393,147 @@ impl Context {
+    ///
+    /// This can subsequently be used to recreate the object in the future.
+    /// The object can only be recreated in a new context, if it was made persistent
+    /// with `evict_control`.
+    ///
+    /// # Arguments
+    /// * `handle` - A handle to the object which should be serialized.
+    ///
+    /// # Returns
+    /// A buffer that can be stored and later deserialized.
+    ///
+    /// # Errors
+    /// * if the TPM cannot serialize the handle, a TSS error is returned.
+    /// * if the buffer length cannot be converted to a `usize`, an `InvalidParam`
+    /// wrapper error is returned.
+    ///
+    /// ```rust
+    /// # use tss_esapi::{
+    /// #     Context, TctiNameConf,
+    /// #     interface_types::resource_handles::Hierarchy,
+    /// #     structures::HashScheme,
+    /// #     utils::create_unrestricted_signing_ecc_public,
+    /// #     interface_types::{
+    /// #         ecc::EccCurve,
+    /// #         algorithm::HashingAlgorithm,
+    /// #         session_handles::AuthSession,
+    /// #     },
+    /// #     structures::EccScheme,
+    /// # };
+    /// # let mut context =
+    /// #     Context::new(
+    /// #         TctiNameConf::from_environment_variable().expect("Failed to get TCTI"),
+    /// #     ).expect("Failed to create Context");
+    /// # context.set_sessions((Some(AuthSession::Password), None, None));
+    /// # let public = create_unrestricted_signing_ecc_public(
+    /// #     EccScheme::EcDsa(HashScheme::new(HashingAlgorithm::Sha256)),
+    /// #     EccCurve::NistP256)
+    /// #     .unwrap();
+    /// let key_handle = context
+    ///     .create_primary(
+    ///         Hierarchy::Owner,
+    ///         public,
+    ///         None,
+    ///         None,
+    ///         None,
+    ///         None,
+    ///     ).unwrap()
+    ///     .key_handle;
+    /// let data = context.tr_serialize(key_handle.into()).unwrap();
+    /// ```
+    pub fn tr_serialize(&mut self, handle: ObjectHandle) -> Result<Vec<u8>> {
+        let mut len = 0;
+        let mut buffer: *mut u8 = null_mut();
+        ReturnCode::ensure_success(
+            unsafe { Esys_TR_Serialize(self.mut_context(), handle.into(), &mut buffer, &mut len) },
+            |ret| {
+                error!("Error while serializing handle: {}", ret);
+            },
+        )?;
+        Ok(to_owned_bytes(
+            buffer,
+            len.try_into().map_err(|e| {
+                error!("Failed to convert buffer len to usize: {}", e);
+                Error::local_error(WrapperErrorKind::InvalidParam)
+            })?,
+        ))
+    }
+
+    /// Deserialize the metadata from `buffer` into a new object.
+    ///
+    /// This can be used to restore an object from a context in the past.
+    ///
+    /// # Arguments
+    /// * `buffer` - The buffer containing the data to restore the object.
+    ///              It can be created using `tr_serialize`.
```

Comment:
`tr_serialize` can be enclosed in square brackets. This has a minor benefit that it creates a clickable link in the documentation.

Reply:
@wiktor-k Thank you, done